python/ray/experimental/dag/class_node.py

```diff
@@ -68,52 +95,71 @@ class ClassMethodNode(DAGNode):
 
     def __init__(
         self,
-        actor: ClassNode,
-        prev_call: Optional["ClassMethodNode"],
         method_name: str,
         method_args: Tuple[Any],
         method_kwargs: Dict[str, Any],
         method_options: Dict[str, Any],
+        other_args_to_resolve: Dict[str, Any],
     ):
+
+        self._bound_args = method_args or []
+        self._bound_kwargs = method_kwargs or {}
+        self._bound_options = method_options or {}
         self._method_name: str = method_name
+        # Parse other_args_to_resolve and assign to variables
+        self._parent_class_node: ClassNode = other_args_to_resolve.get(
+            "parent_class_node"
+        )
+        # Used to track lineage of ClassMethodCall to preserve deterministic
+        # submission and execution order.
+        self._prev_class_method_call: Optional[
+            ClassMethodNode
+        ] = other_args_to_resolve.get("prev_class_method_call", None)
         # The actor creation task dependency is encoded as the first argument,
         # and the ordering dependency as the second, which ensures they are
         # executed prior to this node.
-        DAGNode.__init__(
-            self,
-            (actor, prev_call) + method_args,
+        super().__init__(
+            method_args,
             method_kwargs,
             method_options,
+            other_args_to_resolve=other_args_to_resolve,
         )
+        # TODO: (jiaodong) revisit constraints on dag INPUT before moving out
+        # of experimental folder
+        has_input_node = self._contain_input_node()
+        if has_input_node:
+            if (
+                len(self.get_args()) != 1
+                or not isinstance(self.get_args()[0], InputNode)
+                or self.get_kwargs() != {}
+            ):
+                raise ValueError(
+                    "InputNode marks the entrypoint of user request to the "
+                    "DAG, please ensure InputNode is the only input to a "
+                    "ClassMethodNode, and NOT used in conjunction with, or "
+                    "nested within other args or kwargs."
```

Comment:
Isn't the point of InputNode to allow mixed input? I think you can just remove this.

Reply:
yep with InputNode we can add more flexibility now